src/simsopt/util/netcdf.py

```diff
@@ -0,0 +1,90 @@
+# coding: utf-8
+
+"""
+The purpose of this module is to hide the details of the 
+netCDF library to allow the user to replace scipy.io.netcdf_file 
+with netCDF4's Dataset class if they need netCDF4 support.
+"""
+
+_all__ = ['netcdf_file']
+
+import logging
+from enum import Enum
+from abc import ABC, abstractmethod
+
+
+class NetCDF(Enum):
+    """Supported netCDF libraries."""
+    netCDF4 = 0
+    scipy = 1
+
+
+logger = logging.getLogger(__name__)
+
+try:
+    from netCDF4 import Dataset
+except ImportError as e:
+    logger.debug(str(e))
+    try:
+        from scipy.io import netcdf_file as scipy_netcdf_file
+    except ImportError as e:
+        # this error is fatal
+        logger.debug(str(e))
+        raise e
+    else:
+        netcdf = NetCDF.scipy
+else:
+    netcdf = NetCDF.netCDF4
+
```

Comment:
We are not using `AbstractNetCDF` to define any concrete classes. It is better to put the class definition in the docstrings of `netcdf_file` function, so users can know how to use the returned netcdf object.